enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,486 @@
+well known list of OCP pre-populated Shared Resource.  That list will be in well known ConfigMaps that resides 
+in the managed CSI Driver namespace.  Components / operators other than Shared Resources that want to create pre-populated
+SharedSecrets or SharedConfigMaps should submit PRs against the Shared Resource CSI Driver Operator Repo to update the 
+corresponding ConfigMap with the necessary information.  This is akin to how we create in-payload image streams via the PRs against the 
+samples operator today.
+
+There will be a "ocp-sharedsecret-list" and "ocp-sharedconfigmap-list" ConfigMap.  The key of an entry is the Shared Resource
+name.  The value should be of the form "namespace:name" in order to capture.  These ConfigMaps will be managed by the
+CVO/CSO infrastructure via there definition in the Shared Resourece CSI Driver Operator assets folder in the CSO repository.
+
+A brief example:
+
+```yaml
+kind: ConfigMap 
+apiVersion: v1
+metadata:
+  name: ocp-sharedsecret-list
+  namespace: openshift-cluster-csi-drivers
+data:
+  openshift-etc-pki-entitlement: openshift-config-managed:etc-pki-entitlement
+```
+
+The "openshift-*" naming restriction is new as compared to the initial tech preview release of Shared Resources in 4.10.
+
+A `ValidatingAdmissionWebhook` will be used to inspect 
+all created Pods, and k8s Pod derived types (Deployments, DaemonSets, etc.) with CSI volume mounts referencing the Shared
+Resource CSI Driver and make sure no read-write Shared Resource CSI Volumes are specified.  In other words, the volume attribute
+'readOnly: true' setting must exist.  Note, the understanding is that there is no "duck typing" in vanilla k8s or OpenShift that we can leverage to
+capture all "Pod Spec'able" API objects.
+
+For Builds, BuildRequests, and BuildConfigs, the existing [Aggregated OpenShfit API Server Build Admission plugin](https://github.com/openshift/openshift-apiserver/blob/master/pkg/build/apis/build/validation/validation.go)
+could be updated to enforce this restriction before the build controller in the OCM gets involved in translating a Build, BuildConfig,
+or BuildRequest into a Build Pod.  The Pod level ValidatingAdmissionWebhook could catch this as well, but catching the 
+error case sooner will provide even better usability.  In particular `oc start-build` invocations would now see the error
+immediately.
+
+The 'readOnly: true' for Pods restriction exists in 4.10, but is only enforced at the CSI driver level.
+
+On the associated `ValidatingWebhookConfiguration`, per guidance from David Eads, which should utilize the label selector
+capabilities to help filter which Pods are addressed by our new `ValidatingAdmissionWebhook`.  Consider the following 
+example:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+...
+webhooks:
+  - name: podsharedresourcereadonlywebhook.openshift.com
+    objectSelector:
```

Comment:
Be sure to add the `failurePolicy` - any timeout/500 errors should be ignored ("fail open")
```suggestion
    failurePolicy: Ignore # Do not fail object creation if this webhook times out or returns 5xx errors
    objectSelector:
```

See https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#validatingwebhook-v1-admissionregistration-k8s-io